machine/plugins/base.py

```diff
@@ -65,7 +65,10 @@ def channels(self) -> Dict[str, Channel]:
         :return: a list of all channels in the Slack workspace, where each channel is a
             :py:class:`~machine.models.channel.Channel` object
         """
-        return self._client.channels
+        try:
+            return self._client.channels
+        except:
+            return None
```

Comment:
Why is this needed? `self._client.channels` should always be there. The code you wrote below should fix the problem you've encountered with a DM channel being unknown, but have you ever encountered the whole `channels` dict being empty? I haven't. I don't want to add checks for errors that are not there.